src/stdune/scanf.ml

```diff
@@ -0,0 +1,13 @@
+module Scanf = Dune_caml.Scanf
+
+let unescaped x =
+  match Scanf.unescaped x with
+  | exception End_of_file -> Error ()
```

Comment:
Why only these two? The doc seems to say:

> Raise Scanf.Scan_failure if the input does not match the format.
> Raise Failure if a conversion to a number is not possible.
> Raise End_of_file if the end of input is encountered while some more characters are needed to read the current conversion specification.
> Raise Invalid_argument if the format string is invalid.

(but I don't know when the other two are raised)

Reply:
I'll catch failure as well. However, I think that we shouldn't catch `Invalid_argument _`. A bad formatting string is just a bug.

Reply:
Of course, this does make the function a bit more dangerous to use as `f` could easily raise `Failure` for other reasons.

Reply:
ping @aalekseyev

Reply:
@aalekseyev is away this week, I'll finish the review for him.

> Of course, this does make the function a bit more dangerous to use as f could easily raise Failure for other reasons.

What about `Scanf.ksscanf`? It seems to be exactly for this case

Reply:
In fact, if we use `ksscanf` I'd say we don't even need to inspect the exception

Reply:
I pushed a commit that uses `ksscanf`. However, I'm unable to accomplish this without this private exception I've defined. Perhaps you can think of a way to do it without this?

Reply:
The method with a private exception looks good to me